services/job/job.go

```diff
@@ -0,0 +1,55 @@
+package models
+
+import "time"
+
+type JobStatus string
+
+const (
+	StatusQueued    JobStatus = "queued"
+	StatusRunning   JobStatus = "running"
+	StatusCompleted JobStatus = "completed"
+	StatusFailed    JobStatus = "failed"
+	StatusCancelled JobStatus = "cancelled"
+)
+
+type JobPriority int
+
+// Easier to compare later on
+const (
+	Low JobPriority = iota
+	Middle
+	High
+)
+
+type JobConstraints struct {
+	MaxRuntime        int      `json:"maxRuntime"`        // e.g in seconds
+	PreferredLocation []string `json:"preferredLocation"` // city, continent?
+}
+
+// Json tags helps by (de-)serializing, json:"id" -> "id":"1234", functionality imported by "encoding/json" package
+// If fields value ist empty/nil (needs pointer) and its tagged with omitempty, json ignores it
+
+type Job struct {
+	JobID                string      `json:"jobId"`
+	JobName              string      `json:"jobName"`
+	UserID               string      `json:"userId"`
+	ImageID              string      `json:"imageId"`
+	ImageName            string      `json:"imageName"`
+	ImageVersion         string      `json:"imageVersion"` // e.g :latest
```

Comment:
Would it make sense to extract a separate struct for the image and reference it here?
```
type ImageRef struct {
    ID          string     `json:"id"`
    Name        string     `json:"name"`
    Version     string     `json:"version"`
}
```

```suggestion
	Image                ImageRef    `json:"image"`
```

Reply:
Good Suggestion, its implemented  by the last commit